llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -414,11 +414,20 @@ static string_vector saveDeviceImageProperty(
       if (ImgPSInfo.SpecConstsMet) {
         // extract spec constant maps per each module
         SpecIDMapTy TmpSpecIDMap;
-        SpecConstantsPass::collectSpecConstantMetadata(*ResultModules[I].get(),
-                                                       TmpSpecIDMap);
+        std::vector<char> DefaultValues;
+        SpecConstantsPass::collectSpecConstantMetadata(
+            *ResultModules[I].get(), TmpSpecIDMap, DefaultValues);
         PropSet.add(
             llvm::util::PropertySetRegistry::SYCL_SPECIALIZATION_CONSTANTS,
             TmpSpecIDMap);
+
+        using SpecDefaultMapTy = std::map<StringRef, std::vector<char>>;
+        SpecDefaultMapTy TmpSpecDefaultMap;
+
+        TmpSpecDefaultMap["all"] = DefaultValues;
+        PropSet.add(
+            llvm::util::PropertySetRegistry::SYCL_SPEC_CONSTANTS_DEFAULT_VALUES,
+            TmpSpecDefaultMap);
```

Comment:
It might make sense to create a convenient `add` method to `PropertySetIO`, which allows to add a single value into a property set without creating any temporary maps.

Reply:
Done